### PR TITLE
Pre stage

### PR DESCRIPTION
**Pull request by IlyaVi**

### CODE REVIEW

**Review by alexstolr — COMMENTED** [review]
@IlyaVi Please see my review at:

https://github.com/bloxapp/ssv-web/pull/586

File: src/app/components/applications/SSV/MyAccount/components/Validator/BulkActions/BulkComponent.tsx [hunk 1]
```diff
@@ -71,41 +79,47 @@ const BulkComponent = () => {
...
     if (currentStep === BULK_STEPS.BULK_ACTIONS) {
       setCurrentStep(BULK_STEPS.BULK_CONFIRMATION);
     } else if (currentStep === BULK_STEPS.BULK_CONFIRMATION && currentBulkFlow === BULK_FLOWS.BULK_EXIT) {
-      res = await conditionalExecutor(condition, async () => await validatorStore.exitValidator(`0x${selectedValidators[0]}`, process.item.operators.map((operator: IOperator) => operator.id)), async () => await validatorStore.bulkExitValidators(selectedValidators.map((publicKey: string) => `0x${publicKey}`), process.item.operators.map((operator: IOperator) => operator.id)));
+      const singleFormattedPublicKey = formatValidatorPublicKey(selectedValidatorKeys[0]);
+      res = await conditionalExecutor(condition, async () => await validatorStore.exitValidator(singleFormattedPublicKey, process.item.operators.map((operator: IOperator) => operator.id)), async () => await validatorStore.bulkExitValidators(selectedValidatorKeys.filter((publicKey: string) => selectedValidators[publicKey].isSelected), process.item.operators.map((operator: IOperator) => operator.id)));
```
Comment (alexstolr) [inline src/app/components/applications/SSV/MyAccount/components/Validator/BulkActions/BulkComponent.tsx:90]: I think the code will be a bit cleaner if we extract the functions const exitSIngleFunc = validatorStore.exitValidator and const exitBulkFunc = validatorStore.bulkExitValidators as variable and then call the function conditionalExecutor(condition, exitSingle, exitBulk))

const exitSingle = async () => await validatorStore.exitValidator(singleFormattedPublicKey, process.item.operators.map((operator: IOperator) => operator.id));
      const exitBulk = async () => await validatorStore.bulkExitValidators(selectedValidatorKeys.filter((publicKey: string) => selectedValidators[publicKey].isSelected), process.item.operators.map((operator: IOperator) => operator.id));
      res = await conditionalExecutor(condition, exitSingle, exitBulk);

**Review by alexstolr — COMMENTED** [review]
Some comments but still not finished

File: src/app/components/applications/SSV/MyAccount/components/Validator/BulkActions/BulkComponent.tsx (hunk 1 above)
Comment (axelrod-blox) [reply src/app/components/applications/SSV/MyAccount/components/Validator/BulkActions/BulkComponent.tsx:90]: done

File: src/app/components/applications/SSV/MyAccount/components/Validator/BulkActions/BulkComponent.tsx [hunk 2]
```diff
@@ -32,7 +33,7 @@ const BULK_FLOWS_CONFIRMATION_DATA = {
 };
 
 const BulkComponent = () => {
-  const [selectedValidators, setSelectedValidators] = useState<string[]>([]);
+  const [selectedValidators, setSelectedValidators] = useState<Record<string, { validator: IValidator, isSelected: boolean }>>({});
```
[the remark this answers is not in the document]
Comment (axelrod-blox) [reply src/app/components/applications/SSV/MyAccount/components/Validator/BulkActions/BulkComponent.tsx:36]: +

**Comment by axelrod-blox** [comment]
https://github.com/bloxapp/ssv-web/pull/594 
review and merge this changes before bush to stage

**Pull request merged**